web_src/css/editor-markdown.css

```diff
@@ -18,8 +18,15 @@
   cursor: pointer;
 }
 
-.combo-markdown-editor .markdown-text-editor {
+.ui.form .combo-markdown-editor textarea.markdown-text-editor,
+.combo-markdown-editor textarea.markdown-text-editor {
   display: block;
   width: 100%;
-  height: 200px;
+  min-height: 200px;
+  max-height: calc(100vh - 200px);
+  resize: none;
```

Comment:
```suggestion
  resize: vertical;
```


We should allow manual resize. If manual resize was done, don't auto resize any more.

Reply:
Done in 9af6a4489bb68c5b220282f9af44cf7621209c2a

Reply:
Wouldn't allow horizontal resize.

Reply:
Sorry I didn't get the point.

You can take a try, the current behavior is good enough. If you think anything is wrong, please help to show an example.

Reply:
Just set `resize: vertical` which will ensure it will only resize vertically. I can test later. I'm not sure if `ResizeObserver` is really necessary. Just set a boolean when the user ever touched the resize, and don't do any automatic resizing after that.

Reply:
I think we do not need to set `resize`.

`ResizeObserver` is necessary, otherwise, if user resize the textarea to a larger size, then they change the content value, then the textarea will shrink to the initial height. By `ResizeObserver`, we can remember that the new height as initial height, then the textarea won't shrink smaller that that.

"Just set a boolean when the user ever touched the resize": it still needs `ResizeObserver` IMO.

Reply:
Pretty certain boolean will be enough.

Reply:
> Pretty certain boolean will be enough.

At the moment, I do not know how to distinguish "user resizing" and "browser resizing" behaviors accurately (even if it could be done, it needs more code).

If you use a boolean, it might be set to true by browser resizing, then the user won't have the "auto resizing"  anymore.

Reply:
I guess there must be some property on the resize event to distinguish.

Reply:
We could also just use https://github.com/github/textarea-autosize which github uses.

Reply:
AFAIK, I haven't found one. 

https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserverEntry

Reply:
Above repo uses `mousemove` to detect user resize. Let's just integrate that module please.

Reply:
That's much more complicated than it should be. I really do not like it.

And it uses the `getComputedStyle` which you have expressed strong objection.

Reply:
Then at least implement the mousemove detection. Also check if pasting correctly resizes.

Reply:
Actually I like current behavior. I am not sure I can get  100% your point. If you want to change the behavior, feel free to edit this PR directly.

Reply:
I find `ResizeObserver` needlessly complicated and potentially error-prone. I would pretty much implement it like that module, minus the `getComputedStyle`, just a simple boolean to track user interaction.

Reply:
Well, I can not fully understand your point. The `autosize` says: `future todo: consider replacing with ResizeObserver`

https://github.com/jackmoore/autosize/blob/1e0229b9bb9f51c2b6a2756d901a023b475fa950/src/autosize.js#L130


`ResizeObserver` is correct and it is the future.

So, feel free to edit this PR for your idea.

Reply:
If you use it, you should at minimum debounce the callback as this will trigger hundrets of times per second and consume unnecessary CPU. Still I see no need.

Reply:
> If you use it, you should at minimum debounce the callback as this will trigger hundrets of times per second and consume unnecessary CPU.

I added `console.log`, I didn't see so. We only obverse on the textarea.

`mousemove` could also be triggered plenty of times when moving mouse.

----

To finish the disagreement, I added `auotsize` package. And leave a `TODO` in comment: 

* TODO: even if we set this, the `autosize` package doesn't support manually resize vertically

Reply:
What's your opinion? Continue using the `autosize` package? Or I use my ResizeObserver (which has no performance problem at all) to improve?

Reply:
Will test. Surprised about the TODO because GitHub UI certainly allows manual resize and uses that package.

Reply:
Ah, no, only `@github/textarea-autosize` supports the manual resize.

Reply:
I tried above package but it shows a different bug where the cursor leaves no padding at the bottom. It also reproduces on GitHub.

<img width="78" alt="Screenshot 2023-04-05 at 22 05 41" src="https://user-images.githubusercontent.com/115237/230197085-eb46b56a-8cd6-491d-b5c2-84b2f41abaa1.png">

One nice feature of the package is their `viewportMarginBottom` option, which when set to a value like 130 will ensure the comment button never goes offscreen during resize.

So I'd suggest either switching to to `@github/textarea-autosize` and live with the cursor padding bug, or go back to the ResizeObserver and try to re-implement the `viewportMarginBottom` behaviour while ensuring the size is calculated correctly so the bottom-padding is visible.

Reply:
Actually I think I will just incorporate that function from the module into our code. It does a few nice things over the `ResizeObserver` method, like downsizing as well when content is deleted. I think I should be able to work out that padding bug, than it'll be perfect.

Reply:
Pushed up my version of the `autosize` function. Changes include:

- Removal of `getComputedStyle` of parent element, no observable change in function
- Add scrolling to bottom when cursor is at the end after an input event. This prevents the padding issue above which is caused by the textarea not completely scrolled.

If we could also get rid of the `getComputedStyle` on the textarea, it would be perfect, but it might not be easy so this is LGMT with it too.

Reply:
> It does a few nice things over the ResizeObserver method, like downsizing as well when content is deleted

ResizeObserver  could also do so without bug, I didn't implement it correctly before, and it doesn't have performance problem.

Since you have implemented the "onresize" event listener, then there is no need to change, I am fine with it.